### ISSUE TITLE
[Bug] Dialog Doesn't Receive Focus When No Textfield is Present

### ISSUE THREAD

**Issue by iamkartiknayak** [pop-os/libcosmic#819]
**Issue/Description:**  
Currently, when a dialog is opened without a textfield inside, it does not receive focus automatically. This can lead to accessibility and usability issues, as users may need to manually click the dialog to interact with it.  

**Steps to Reproduce:**  
- Open COSMIC Edit.  
- Type something then try to close
- It'll show a dialog to save before closing the tab
- Observe that the dialog does not receive focus automatically, making it practically unusable.  

**Expected Behavior:**  
The dialog should receive focus automatically when opened, even if no text-field is present, ensuring proper accessibility and interaction.  

**Other Notes:**
When pressing tab it moved the text instead of moving focus between the buttons in the dialog

<img src="https://github.com/user-attachments/assets/d37ad222-e910-47d0-b335-bab4347e600b" width=600>

**Comment by iamkartiknayak** [comment]
It would be great if when the dialog is active user can no longer interact with the parent window in the bg

**Comment by mmstick** [comment]
Dialogs created by `Application::dialog` do define themselves as modal if using the default implementation of `Application::view_main`. https://github.com/pop-os/libcosmic/blob/cb682be3c887bcbf591309d350bac4d4710e104b/src/app/mod.rs#L791

**Comment by mmstick** [comment]
Partially fixed by #832

**Comment by ImanSeyed** [comment]
Any updates on this? I believe pop-os/cosmic-files#983 is also related to this issue.

**Comment by griest024** [comment]
Should the dialog trap focus?

**Comment by nathanpainchaud** [comment]
Is there an update on this? I think it's a pretty fundamental feature that people expect to work properly in a DE.

It also affects a lot of downstream components. https://github.com/pop-os/cosmic-files/issues/983 was already mentioned above, and a quick search of the repo for cosmic-edit shows https://github.com/pop-os/cosmic-edit/issues/494 and https://github.com/pop-os/cosmic-edit/issues/210 related to focus remaining in editor text field when save/close dialog box is shown.

There are probably a lot more duplicate/similar issues in various COSMIC components that could be traced back to here...

**Comment by mmstick** [comment]
@nathanpainchaud This issue will be updated when there is an update. It's better to subscribe and upvote an issue.

**Comment by griest024** [comment]
@mmstick should we have a master list of all of the dialog focus issues? It does seem to still be fairly pervasive even though https://github.com/pop-os/cosmic-files/issues/983 does appear to be fixed.

**Comment by mmstick** [comment]
No meta issue necessary, but duplicates can be pointed here.

**Comment by nathanpainchaud** [comment]
> It's better to subscribe and upvote an issue.

@mmstick Already did 😉 

I just wanted to point out that this issue might be more important than it appears based on upvotes. Many end users will open issues specific to their use-cases, and won't take the time to trace back their problems to this underlying issue to upvote it (so that it hopefull gets prioritized).
